hadoop-hdds/common/src/test/java/org/apache/hadoop/ozone/audit/TestOzoneAuditLogger.java

```diff
@@ -158,20 +184,34 @@ private void verifyLog(String expected) throws IOException {
       }
       i++;
     }
-
-    // When log entry is expected, the log file will contain one line and
-    // that must be equal to the expected string
-    assertTrue(lines.size() != 0);
-    assertTrue(expected.equalsIgnoreCase(lines.get(0)));
+    //check if every expected string can be found in the log entry
+    for(String expected : expectedStrings) {
+      assertTrue(contains(lines, expected));
```

Comment:
instead of writing a contains method we can use the following combination of assertThat and matchers to gain the same functionality:
```java
assertThat(lines, hasItem(containsString(expected)));
```

If we change to check that lines starts with the lines specified in expectedStrings in order, then an other approach like this solves the problem:
```java
assertThat(
    lines.subList(0,expectedStrings.length),
    containsInOrder(expectedStrings)
);
```

where containsInOrder is defined as:
```java
private Matcher<Iterable<? extends String>> containsInOrder(
    String[] expectedStrings) {
  return IsIterableContainingInOrder.contains(
      Arrays.stream(expectedStrings)
          .map(str -> containsString(str))
          .collect(Collectors.toList())
  );
}
```

In order for this to work it is necessary to add org-harmcrest:harmcrest-all as a test scope dependency to the hdds-common module, and change the messageIncludesMultilineException test, to call verifyLog as:
```java
verifyLog(
    "ERROR | OMAudit | user=john | ip=192.168.0.1 | op=CREATE_VOLUME " +
        "{key1=value1, key2=value2} | ret=FAILURE",,
    "TestException: " + exceptionMsg,
    "org.apache.hadoop.ozone.audit."
        + "TestOzoneAuditLogger."
        + "messageIncludesMultilineException"
);
```
assuming that the message text given to the TestException is extracted to a local variable called exceptionMsg ;)

Reply:
Added hamcrest suggestion. I find the legibility of hamcrest messages useful especially for evaluating log entries